pkg/helm/helm.go

```diff
@@ -106,107 +89,177 @@ func New(opts Options) (*Client, error) {
+	if err != nil {
+		return Release{}, err
 	}
 
-	if err := valuesFile.Close(); err != nil {
-		return nil, err
+	var release Release
+	if err := json.Unmarshal(stdout, &release); err != nil {
+		return Release{}, err
 	}
+	return release, nil
+}
 
-	flags := []string{
-		fmt.Sprintf("--namespace=%s", o.Namespace),
-		fmt.Sprintf("--values=%s", valuesFile.Name()),
+func (c *Client) ShowValues(ctx context.Context, chart string, values any) error {
+	stdout, _, err := c.runHelm(ctx, "show", "values", chart)
+	if err != nil {
+		return err
 	}
+	return yaml.Unmarshal(stdout, &values)
+}
 
-	if o.CreateNamespace {
-		flags = append(flags, "--create-namespace")
+func (c *Client) GetValues(ctx context.Context, release *Release, values any) error {
+	stdout, _, err := c.runHelm(ctx, "get", "values", release.Name, "--output=json", "--namespace", release.Namespace)
+	if err != nil {
+		return err
 	}
+	return json.Unmarshal(stdout, &values)
+}
 
-	if o.Name == "" {
-		flags = append(flags, "--generate-name")
-	}
+type InstallOptions struct {
+	CreateNamespace bool   `flag:"create-namespace"`
+	Name            string `flag:"-"`
+	Namespace       string `flag:"namespace"`
+	Values          any    `flag:"-"`
+	Version         string `flag:"version"`
+	NoWait          bool   `flag:"wait"`
+	NoWaitForJobs   bool   `flag:"wait-for-jobs"`
+	GenerateName    bool   `flag:"generate-name"`
+	ValuesFile      string `flag:"values"`
+}
 
-	if o.Version != "" {
-		flags = append(flags, fmt.Sprintf("--version=%s", o.Version))
+func (c *Client) Install(ctx context.Context, chart string, opts InstallOptions) (Release, error) {
+	if opts.Name == "" {
+		opts.GenerateName = true
 	}
 
-	if !o.NoWait {
-		flags = append(flags, "--wait")
+	if opts.Values != nil {
+		var err error
+		opts.ValuesFile, err = c.writeValues(opts.Values)
+		if err != nil {
+			return Release{}, err
+		}
 	}
 
-	if !o.NoWaitForJobs {
-		flags = append(flags, "--wait-for-jobs")
-	}
+	args := []string{"install", chart, "--output=json"}
+	args = append(args, ToFlags(opts)...)
 
-	return flags, nil
-}
+	if opts.Name != "" {
+		args = slices.Insert(args, 1, opts.Name)
+	}
 
-func (c *Client) List(ctx context.Context) ([]Release, error) {
-	stdout, _, err := c.runHelm(ctx, "list", "-A", "--output=json")
+	stdout, _, err := c.runHelm(ctx, args...)
 	if err != nil {
-		return nil, err
+		return Release{}, err
 	}
 
-	var releases []Release
-	if err := json.Unmarshal(stdout, &releases); err != nil {
-		return nil, err
+	// TODO(chrisseto): The result of `helm install` appears to be its own
+	// unique type. The closest equivalent is `helm get all` but that can't be
+	// output as JSON.
+	// For now, we scrape out the name and use `helm get metadata` to return
+	// consistent information.
+	var result map[string]any
+	if err := json.Unmarshal([]byte(stdout), &result); err != nil {
+		return Release{}, err
 	}
```

Comment:
I'm not sure why attempt to unmarshal `stdout` is done if the Release will be provided by `helm get metadata` output.